prototypes/emdpg_waveguide/EM_waveguide.cc

```diff
@@ -0,0 +1,2451 @@
+
+    // Default frequency and excitation modes
+    frequency = 670356315.23; // This corresponds to the TE10 mode with a
+                              // wavelenght of 1 for a = 0.25
+    mode_x = 1;               // Mode in x direction
+    mode_y = 0;               // Mode in y direction
+
+    add_parameter("frequency", frequency, "Frequency of the signal");
+
+    add_parameter("mode x", mode_x, " TE mode in x direction");
+
+    add_parameter("mode y", mode_y, " TE mode in y direction");
+
+    // Default mesh parameters
+    n_refinements        = 0;
+    initial_refinements  = 0;
+    n_repetitions        = {1, 1, 4};
+    adaptive_refinement  = false;
+    max_refinement_level = 8;
+
+    add_parameter("n refinements",
+                  n_refinements,
+                  "Number of refinements cycle for mesh refinement");
+
+    add_parameter("initial refinements",
+                  initial_refinements,
+                  "Number of refinements for the coarse mesh before starting");
+
+    add_parameter("n repetitions",
+                  n_repetitions,
+                  "Number of repetitions for the subdivided hyper rectangle");
+
+    add_parameter("adaptive refinement",
+                  adaptive_refinement,
+                  "If true, the code will use adaptive mesh refinement, "
+                  "otherwise it will use uniform mesh refinement");
+
+    add_parameter("max refinement level",
+                  max_refinement_level,
+                  "Maximum number of refinements from the coarse mesh");
+
+    // Default finite element parameters
+    degree = 0;
+    add_parameter("degree", degree, "order of the DGQ finite element space");
+
+    delta_degree = 1;
+    add_parameter("delta degree",
+                  delta_degree,
+                  "order difference between the trial and test space");
+  }
+
+  /* Create analytical solution class for the electric field (E).
+   * The corresponding solution for the electric field follows the physicist's
+   * time convention with the imaginary unit in the exponent being negative
+   * (i.e. exp(-i wt) ). The function manage both the real and complex
+   * components of the solution. Since the frequency, the material properties,
+   * and the geometry are arguments for the parameter file, they are used in the
+   * constructor to initialize the different wavenumber. Note that everything is
+   * addimensionalize using the following : \tilde{\mathbf{E}} =
+   * \frac{\mathbf{E}}{E_0}, \tilde{\mathbf{H}} = \frac{Z_0}{E_0} \mathbf{H},
+   *       \varepsilon_r = \frac{\varepsilon}{\varepsilon_0},
+   *       \mu_r = \frac{\mu}{\mu_0},
+   *       Z_r = \frac{Z}{Z_0},
+   *       \sigma_r = \frac{\sigma}{\omega \varepsilon_0},
+   *       \tilde{\omega} =  \frac{\omega}{k_0 c_0},
+   *       \tilde{\mathbf{J}} = \frac{1}{\omega \varepsilon E_0 } \mathbf{J}
+   *       \tilde{k} = L \mathbf{k},
+   *       \tilde{\nabla} = L \nabla.
+   */
```

Comment:
if you want the math expressions to appear correctly in the doxygen, you need to add them between \f$ \f$.